.dev/compare_branches.R

```diff
@@ -0,0 +1,230 @@
+#!/usr/local/bin/Rscript
+
+# compare the lints obtained before/after a given PR/branch vs current master
+
+library(optparse)
+library(dplyr)
+library(purrr)
+library(tibble)
+library(usethis)
+library(gert)
+library(devtools)
+
+if(!file.exists("lintr.Rproj")) {
+  "compare_branches.R should be run inside the lintr-package directory"
+}
+
+param_list <- list(
+  optparse::make_option(
+    "--linters",
+    default = "object_usage_linter",
+    help = "Run the comparison for these linter(s) (comma-separated) [default %default]"
+  ),
+  optparse::make_option(
+    "--branch",
+    default = if (interactive()) {
+      readline("Name a branch to compare to master (or skip to enter a PR#): ")
+    },
+    help = "Run the comparison for master vs. this branch"
+  ),
+  optparse::make_option(
+    "--pr",
+    default = if (interactive()) {
+      as.integer(readline("Name a PR # to compare to master (skip if you've entered a branch): "))
+    },
+    type = "integer",
+    help = "Run the comparison for master vs. this PR"
+  ),
+  optparse::make_option(
+    "--packages",
+    default = if (interactive()) {
+      readline("Provide a comma-separated list of packages (skip to provide a directory): ")
+    },
+    help = "Run the comparison using these packages (comma-separated)"
+  ),
+  optparse::make_option(
+    "--pkg_dir",
+    default = if (interactive()) {
+      readline("Provide a directory where to select packages (skip if already provided as a list): ")
+    },
+    help = "Run the comparison using all packages in this directory"
+  ),
+  optparse::make_option(
+    "--sample_size",
+    type = "integer",
+    default = if (interactive()) {
+      as.integer(readline("Enter the number of packages to include (skip to include all): "))
+    },
+    help = "Select a sample of this number of packages from 'packages' or 'pkg_dir'"
+  ),
+  optparse::make_option(
+    "--outfile",
+    default = file.path("~", sprintf("lintr_compare_branches_%d.csv", as.integer(Sys.time()))),
+    help = "Destination file to which to write the output"
+  )
+)
+
+params <- optparse::parse_args(optparse::OptionParser(option_list = param_list))
+# treat any skipped arguments from the prompt as missing
+if (interactive()) {
+  for (opt in c("branch", "pr", "packages", "pkg_dir", "sample_size")) {
+    if (params[[opt]] == "") params[[opt]] = NULL
+  }
+}
+
+linter_names <- strsplit(params$linters, ",", fixed = TRUE)[[1L]]
+
+# prioritize "branch"
+is_branch <- FALSE
+if (!is.null(params$branch)) {
+  branch <- params$branch
+  is_branch <- TRUE
+} else if (!is.null(params$pr)) {
+  pr <- params$pr
+} else {
+  message("Please supply a branch (--branch) or a PR number (--pr)")
+  q("no")
```

Comment:
Why not `stop()`? `q()` in interactive seems like a bad idea.

Reply:
agree